autoscheduler/frontend/src/redux/actions/schedules.ts

```diff
@@ -47,3 +55,74 @@ export function renameSchedule(index: number, name: string): RenameScheduleActio
     name,
   };
 }
+
+export function generateSchedules(includeFull: boolean):
+ThunkAction<Promise<void>, RootState, undefined, ReplaceSchedulesAction | SelectScheduleAction> {
+  return async (dispatch, getState): Promise<void> => {
+    const { courseCards, availability, term } = getState();
+
+    const checkIfEmpty = (schedules: Meeting[][]): Meeting[][] => {
+      if (schedules.length === 0) throw Error('No schedules found. Try widening your criteria.');
+      return schedules;
+    };
+
+    // make courses object
+    const courses = [];
+    for (let i = 0; i < courseCards.numCardsCreated; i++) {
+      if (courseCards[i] && courseCards[i].course) {
+        const courseCard = courseCards[i];
+
+        // Iterate through the sections and only choose the ones that are selected
+        const selectedSections = courseCard.sections
+          .filter((sectionSel) => sectionSel.selected)
+          .map((sectionSel) => sectionSel.section.sectionNum);
+
+        const [subject, courseNum] = courseCard.course.split(' ');
+        const isBasic = courseCard.customizationLevel === CustomizationLevel.BASIC;
+
+        // The default option for honors and web when the Section customization level is selected
+        const filterDefault = 'no_preference';
+
+        courses.push({
+          subject,
+          courseNum,
+          sections: isBasic ? [] : selectedSections, // Only send if "Section" customization level
+          // Only send if "Basic" level
+          honors: isBasic ? (courseCard.honors ?? filterDefault) : filterDefault,
+          web: isBasic ? (courseCard.web ?? filterDefault) : filterDefault,
+        });
+      }
+    }
+    // make availabilities object
+    const availabilities = availability.map((avl) => ({
+      startTime: formatTime(avl.startTimeHours, avl.startTimeMinutes, true, true).replace(':', ''),
+      endTime: formatTime(avl.endTimeHours, avl.endTimeMinutes, true, true).replace(':', ''),
+      day: avl.dayOfWeek,
+    }));
+
+    // make request to generate schedules and update redux, will also save availabilities
+    return fetch('scheduler/generate', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRFToken': Cookies.get('csrftoken'),
+      },
+      body: JSON.stringify({
+        term,
+        includeFull,
+        courses,
+        availabilities,
+      }),
+    }).then(
+      (res) => res.json(),
+    ).then(
+      (generatedSchedules: any[][]) => generatedSchedules.map(parseAllMeetings),
+    ).then(
+      checkIfEmpty,
+    )
+      .then((schedules: Meeting[][]) => {
+        dispatch(replaceSchedules(schedules));
+        dispatch(selectSchedule(0));
+      });
+  };
```

Comment:
In case something else goes wrong, it would be nice to catch an error at the end here with something like "There was an error generating schedules, please try again"

Reply:
Actually nevermind I did this in my branch for generate schedules error messages